### PR TITLE
Allow apprunner mojo to override application config

### PR DESCRIPTION
**Pull request by laurentgo**
Add `applicationProperties` configuration to Apprunner mojo to override
application configuration for a given execution.

Fixes projectnessie/nessie#304

---
This change is [Reviewable](https://reviewable.io/reviews/projectnessie/nessie/306)

### CODE REVIEW

File: tools/apprunner-maven-plugin/src/main/java/com/dremio/nessie/quarkus/maven/QuarkusAppStartMojo.java [hunk 1]
```diff
@@ -70,6 +73,12 @@
   @Parameter(property = "nessie.apprunner.appArtifactId", required = true)
   private String appArtifactId;
 
+  /**
+   * Application configuration properties.
+   */
+  @Parameter
```
Comment (rymurr) [inline tools/apprunner-maven-plugin/src/main/java/com/dremio/nessie/quarkus/maven/QuarkusAppStartMojo.java:79]: This is used via a `<properties>` block in the config of the plugin then? And its a user error if the properties are invalid or contracticting

File: tools/apprunner-maven-plugin/src/main/java/com/dremio/nessie/quarkus/maven/QuarkusApp.java [hunk 2]
```diff
@@ -62,7 +67,8 @@ private QuarkusApp(RunningQuarkusApplication runningApp) {
    * @throws MojoExecutionException if an error occurs during execution
    */
   public static QuarkusApp newApplication(MavenProject project, RepositorySystem repoSystem,
-      RepositorySystemSession repoSession, String appArtifactId)
+      RepositorySystemSession repoSession, String appArtifactId, Properties applicationProperties,
+      Collection<Path> additionalDependencyPaths)
```
Comment (rymurr) [inline tools/apprunner-maven-plugin/src/main/java/com/dremio/nessie/quarkus/maven/QuarkusApp.java:71]: what are the additional dependencies for?

**Review by rymurr — COMMENTED** [review]
looks good to me. I am not sure I understand yet how to pass these configs to the underlying tests

File: tools/apprunner-maven-plugin/src/main/java/com/dremio/nessie/quarkus/maven/QuarkusAppStartMojo.java (hunk 1 above)
Comment (laurentgo) [reply tools/apprunner-maven-plugin/src/main/java/com/dremio/nessie/quarkus/maven/QuarkusAppStartMojo.java:79]: the configuration key is derived from the field name, so in this case, it's `applicationProperties`. They take precedence over values defined in `application.properties`

File: tools/apprunner-maven-plugin/src/main/java/com/dremio/nessie/quarkus/maven/QuarkusApp.java (hunk 2 above)
Comment (laurentgo) [reply tools/apprunner-maven-plugin/src/main/java/com/dremio/nessie/quarkus/maven/QuarkusApp.java:71]: The `MojoSourceConfig` needs to be part of the application runtime in order to be able to set the properties. Should probably find a better/smarter way of (it's still a bit experimental)

**Review by jacques-n — APPROVED** [review]

**Review by rymurr — APPROVED** [review]

**Pull request merged**